enumerable_blocks.rb

```diff
@@ -0,0 +1,157 @@
+
+    element = to_a
+    k = 0
+    while k < element.length
+      yield element[k], k
+      k += 1
+    end
+    self
+  end
+
+  # my_select
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    my_each { |i| arr << i if yield i }
+    arr
+  end
+
+  # my_all?
+  def my_all?(argument = nil)
+    if block_given? && !argument.nil?
+      return false unless my_all?(argument) && my_all?
+    end
+    if block_given?
+      my_each { |item| return false if yield(item) }
+    elsif !block_given? && argument.nil?
+      my_each { |_item| return true unless argument == false || argument.nil? }
+    elsif !block_given? && argument.nil?
+      my_each { |item| return false if item == false || item.nil? }
+    elsif argument.is_a?(Class)
+      my_each { |_item| return false unless argument.is_a?(Class) }
+    elsif argument == Regexp
+      my_each { |item| return true if item.match(Regexp) }
+    elsif argument == Regexp
+      my_each { |item| return false unless item.match(Regexp) }
+    else
+      my_each { |item| return false if item != argument }
+    end
+    true
+  end
+
+  # my_any?
+  def my_any?(argument = nil)
+    if block_given? && !argument.nil?
+      return true if my_any?(argument) && my_any?
+    end
+    if block_given? && !argument.nil?
+      my_each { |item| return true if yield(item) }
+    elsif argument.nil?
+      my_each { |item| return true if item }
+    elsif argument.is_a?(Class)
+      my_each { |item| return true if item.instance_of?(argument) }
+    elsif argument.instance_of?(Regexp)
+      my_each { |item| return true if item.match(argument) }
+    else
+      my_each { |item| return true if item == argument && item.class <= argument.class }
+    end
+    false
+  end
+
+  # my_none?
+  def my_none?(argument = nil)
+    if block_given? && !argument.nil?
+      return true if my_none?(argument) && my_none?
+    end
+    if block_given?
+      my_each { |item| return false if yield(item) }
+    elsif !block_given? && argument.nil?
+      my_each { |item| return false if item }
+    elsif argument.is_a?(Class)
+      my_each { |_item| return true unless argument.is_a?(Class) }
+
+    elsif argument.is_a?(Numeric)
+      my_each { |_item| return true unless argument.is_a?(Numeric) }
+
+    elsif argument.is_a?(Class)
+      my_each { |item| return false if item.is_a?(Class) }
+
+    elsif argument == Regexp
+      my_each { |_item| return true unless argument.match(Regexp) }
+
+    elsif argument == Regexp
+      my_each { |item| return false if item.match(Regexp) }
+
+    else
+      my_each { |item| return false if item == argument && item.class <= argument.class }
+    end
+    true
+  end
+
+  def my_count(param = nil)
+    count = 0
+    if !block_given? && param.nil?
+      count = length
+    elsif block_given?
+      my_each { |element| count += 1 if yield element }
+    else
+      count = my_select { |element| element == param }.length
+    end
+    count
+  end
```

Comment:
# my_count threw an error when I ran this code `puts (5...50).count`
:point_down: Read the error thrown by `my_count`
```ruby
Traceback (most recent call last):
        1: from enumerable_blocks.rb:159:in `<main>'
enumerable_blocks.rb:117:in `my_count': undefined local variable or method `length' for 5...50:Range (NameError)
```